### DISCUSSION TITLE
Transmitting data from motion capture to crazyflie

### DISCUSSION THREAD

**Discussion by cc299792458** [bitcraze/discussions#246] [Q&A] [answered]
Hi, Bitcraze!

Sorry to bother you. I have encountered some simple questions when I tried to transmit data got from motion capture(Vicon) to crazyflie.
I use function 'cf.extpos.send_pose()' to send the data. However, when I use log to check 'stateEstimate.x', 'stateEstimate.y' and 'stateEstimate.z', they are 'nan'.
Besides, I also checked 'stabilizer.roll', 'stabilizer.pitch' and 'stabilizer.yaw' at the same time. Althought they are not 'nan', I found them were very large(such as 50, which is impossible for radian, I think). So I guess there must be some problem in rotation data, too.
I follow the example script 'basiclog.py' to set up my logger. 
Before sending data to crazyflie, I just activate kalman estimator, set 'locSrv.extQuatStdDev' to 0.06 and do nothing else.

I wonder is there anything I forget to do or do in a wrong way?

Any help would be appreciate!

**Comment by krichardsson** [comment, marked as answer]
> I use function 'cf.extpos.send_pose()' to send the data. However, when I use log to check 'stateEstimate.x', 'stateEstimate.y' and 'stateEstimate.z', they are 'nan'.

NaN sounds like something is wrong, I would not expect that. Have you tried to set up the logging without setting any parameters or pushing any data to the Crazyflie? Do you get the same result? Have you tried to set up logging with the client to see what you get?

> I also checked 'stabilizer.roll', 'stabilizer.pitch' and 'stabilizer.yaw' at the same time. Althought they are not 'nan', I found them were very large(such as 50, which is impossible for radian, I think). So I guess there must be some problem in rotation data, too.

Roll, pitch and yaw are in degrees so 50 makes sense, [documentation here](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#stateestimate)

**Reply by cc299792458** [reply]
Hi!
Thank you for your help and sorry for replying late.
I try some other cases and find the question maybe related to the rotation data.
I used to send the rotation data in maybe over 10 digits. When I try to maintain only 6 digits of it and send the data to crazyflie, I can get the correct feedback(log data) from crazyflie.
Though the question is solved, I haven't figured out the reason of the phenomenon. 
Thank you again for your help!

**Reply by krichardsson** [reply]
Great that it worked out!

**Reply by cc299792458** [reply]
Hi, I finally know the reason! Vicon will reture rotation of qx, qy, qz, qw = 0, 0, 0, 0 at start. When we use extpos.send_pose() to send that data to crazyflie, it will cause nan. So just skip the sigularity data and continue, that will be fine!

**Reply by krichardsson** [reply]
Great!
Good to know that there might be singularities.
